### PR TITLE
Fall back to tools restored for .NET Core 1.x if they haven't been restored for .NET Core 2.0

### PR DESCRIPTION
**Pull request by dsplaisted**
NuGet restore in Visual Studio doesn't look at the `DotnetCliToolTargetFramework`, so it may restore tools for netcoreapp1.0.  So if that happens, this PR makes the CLI fall back to looking for a netcoreapp1.0 or netcoreapp1.1 tool restore.

@nguerrera @livarcocc

### CODE REVIEW

**Comment by dsplaisted** [comment]
@MattGertz for approval.  This fixes some scenarios where Visual Studio's NuGet restore is used, but a command is run with the 2.0 CLI.

**Comment by dsplaisted** [comment]
@dotnet-bot test Ubuntu x64 Release Build

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsCommandResolver.cs [hunk 1]
```diff
@@ -141,9 +141,33 @@ private CommandSpec ResolveFromProjectTools(CommandResolverArguments commandReso
...
+
+            //  NuGet restore in Visual Studio may restore for netcoreapp1.0.  So if that happens, fall back to
+            //  looking for a netcoreapp1.0 or netcoreapp1.1 tool restore.
             if (toolLockFile == null)
             {
-                return null;
+                if (toolPackageFramework.Framework == FrameworkConstants.FrameworkIdentifiers.NetCoreApp &&
+                    toolPackageFramework.Version >= new Version(2, 0, 0))
+                {
+                    foreach (var fallbackFramework in new [] { "netcoreapp1.1", "netcoreapp1.0"})
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsCommandResolver.cs]: Should we just add project.DotnetCliToolTargetFramework to this array and remove the GetToolLockFile call above?

Comment (dsplaisted) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectToolsCommandResolver.cs]: Yeah, good point.

**Pull request merged**